omnigibson/object_states/particle_modifier.py

```diff
@@ -254,52 +272,64 @@ def overlap_callback(hit):
         if self.method == ParticleModifyMethod.PROJECTION:
             # Construct naming prefix to apply to generated prims
             name_prefix = f"{self.obj.name}_{self.__class__.__name__}"
-            # Make sure projection mesh params are specified
-            # Import here to avoid circular imports
-            from omnigibson.objects.dataset_object import DatasetObject
-            if self._projection_mesh_params is None and isinstance(self.obj, DatasetObject):
-                # We try to grab metadata for this object
-                self._projection_mesh_params = self.obj.metadata.get("meta_links", dict()).get(m.LINK_NAME, None)
-            # Sanity check to make sure projection mesh params is not None
-            assert self._projection_mesh_params is not None, \
-                f"Projection mesh params must be specified for {self.obj.name}'s {self.__class__.__name__} state " \
-                f"when method=ParticleModifyMethod.PROJECTION!"
-
-            mesh_prim_path = f"{self.link.prim_path}/projection_mesh"
+            shape_defaults = {
+                "radius": 0.5,
+                "height": 1.0,
+                "size": 1.0,
+            }
+            mesh_prim_path = f"{self.link.prim_path}/mesh_0"
             # Create a primitive shape if it doesn't already exist
-            radius, height = self._projection_mesh_params["extents"][0] / 2.0, self._projection_mesh_params["extents"][2]
-            if not get_prim_at_path(mesh_prim_path):
+            pre_existing_mesh = get_prim_at_path(mesh_prim_path)
+            if not pre_existing_mesh:
+                # Projection mesh params must be specified in order to determine scalings
+                assert self._projection_mesh_params is not None, \
+                    f"Must specify projection_mesh_params for {self.__class__.__name__} " \
+                    f"since it has no pre-existing projection mesh!"
                 mesh = UsdGeom.__dict__[self._projection_mesh_params["type"]].Define(og.sim.stage, mesh_prim_path).GetPrim()
-                mesh.GetAttribute("height").Set(height)
-                mesh.GetAttribute("radius").Set(radius)
+                property_names = set(mesh.GetPropertyNames())
+                for shape_attr, default_val in shape_defaults.items():
+                    if shape_attr in property_names:
+                        mesh.GetAttribute(shape_attr).Set(default_val)
+            else:
+                # Potentially populate projection mesh params if the prim exists
+                if self._projection_mesh_params is None:
+                    self._projection_mesh_params = {
+                        "type": pre_existing_mesh.GetTypeName(),
+                        "extents": np.array(pre_existing_mesh.GetAttribute("xformOp:scale").Get()),
+                    }
```

Comment:
So if pre_exsiting_mesh is True, does it take priority over `self._projection_mesh_params`? I thought it was the opposite based on our last discussion. I don't have a strong opinion on this tho but I just want to point this out.

Reply:
Nope -- that's why there's the if statement in L295 -- it only gets populated by the externally-defined mesh if there's no params specified.

Note that this guarantees that `projection_mesh_params` are always populated, so that a few lines later (L315) it can be used to set the projection mesh scale

Reply:
remove mesh if type mismatch beforehand

Reply:
result: can't delete mesh, instead, assert if projection_mesh_params type and pre-existing mesh type don't match